src/pybind/mgr/cephadm/templates/services/keepalived/keepalived.conf.j2

```diff
@@ -0,0 +1,35 @@
+# {{ cephadm_managed }}
+vrrp_script check_haproxy {
+    script "curl http://localhost:{{ spec.ha_proxy_port }}/haproxy_test"
+    weight -20
+    interval 2
+    rise 2
+    fall 2
+}
+
+vrrp_instance VI_0 {
+  state {{ state }}
+  priority 100
+  interface {{ spec.virtual_ip_interface }}
+  virtual_router_id 51
+  advert_int 1
+  authentication {
+      auth_type PASS
+      auth_pass {{ spec.keepalived_password }}
+  }
+  unicast_src_ip {{ host_ip }}
+  unicast_peer {
+    {% for ip in other_ips  %}
+    {{ ip }}
+    {% endfor %}
+  }
+  virtual_ipaddress {
+    {{ spec.virtual_ip_address }}/24 dev {{ spec.virtual_ip_interface }}
```

Comment:
Is this "/24" supposed to be added explicitly to the end of the virtual ip address here or is that meant to be part of the virtual ip address arg the user provides? In the haproxy config this "/24" isn't added explicitly

Reply:
Yes lets include the network mask in the "variable" part. In this way is more flexible for the final user. 

change to:
`{{ spec.virtual_ip_address }} dev {{ spec.virtual_ip_interface }}`

It will work if we use as virtual_ip_address just simple ip (192.168.1.150/) ,,.. but it also will work wtih things like : 192.168.1.150/24 , 192.168.1.150/32, etc ..